pkg/proxy/nftables/proxier_test.go

```diff
@@ -507,27 +507,20 @@ func TestOverallNFTablesRules(t *testing.T) {
 		add chain ip kube-proxy filter-forward { type filter hook forward priority -101 ; }
 		add rule ip kube-proxy filter-forward ct state new jump endpoints-check
 		add rule ip kube-proxy filter-forward jump forward
-		add rule ip kube-proxy filter-forward ct state new jump firewall-check
 		add chain ip kube-proxy filter-input { type filter hook input priority -101 ; }
 		add rule ip kube-proxy filter-input ct state new jump endpoints-check
-		add rule ip kube-proxy filter-input ct state new jump firewall-check
+		add rule ip kube-proxy firewall-HVFWP5L3-ns5/svc5/tcp/p80 ip saddr { 203.0.113.0/25 } goto external-HVFWP5L3-ns5/svc5/tcp/p80 comment "LoadBalancer traffic allowed by LoadBalancerSourceRanges"
```

Comment:
Rather than adding a firewall chain, you could just add this rule to the external traffic chain. [We can't do that in iptables because it would possibly be a breaking change.](https://github.com/kubernetes/kubernetes/blob/v1.29.0/pkg/proxy/iptables/proxier.go#L1167-L1169) But that's one of the things we talked about changing in nftables, and if it makes the code simpler, then let's do it.

(Also, in that case, maybe instead of using an anonymous set, you could have a named set? Then you don't need the extra regexp stuff for parsing anonymous sets in the packet tracer...)

Reply:
> the field is not even configurable when the type is not **LoadBalancer**, would it look a bit inconsistent?

Right, if you had a service of `type: NodePort` then  you wouldn't be able to configure source ranges. The idea is that if you have a service of `type: LoadBalancer` on a cloud platform where the load balancers redirect traffic to the NodePorts, then source ranges ought to apply to the NodePort traffic in that case, since the NodePort traffic is really load balancer traffic.

But anyway, externalIPs _shouldn't_ be subject to source ranges, so we can't just do that from the `externalTrafficChain`...

> I thought in either case we could use either anonymous set or named set?

Yes, you can, it's just that creating a new chain _and_ a new set for something so small seems excessive, but if you aren't creating a new chain, then creating a new set doesn't seem as bad?

(I kind of don't like having to parse the anonymous set in `helpers_test.go`, though it's not really _that_ bad.)

Reply:
Got it, then I could change it to a named set if we end up with not needing the firewall chain.